src/main/java/edu/cornell/kfs/pmw/batch/businessobject/PaymentWorksVendor.java

```diff
@@ -221,6 +219,14 @@ public void setRequestingCompanyLegalLastName(String requestingCompanyLegalLastN
         this.requestingCompanyLegalLastName = requestingCompanyLegalLastName;
     }
 
+    public String getRequestingCompanyLegalNameForProcessing() {
+        if (StringUtils.isNotBlank(requestingCompanyLegalName)) {
+            return requestingCompanyLegalName;
+        } else {
+            return requestingCompanyLegalLastName + VendorConstants.NAME_DELIM + requestingCompanyLegalFirstName;
+        }
+    }
+
```

Comment:
I agree that this logic needs to be consolidated somewhere. However, because of where this is needed within the BOs and DTOs, it may not be practical to have them use the batch utility service directly. I may have to create a separate utility class to handle this, and possibly have it defer to the batch utility service. This would make it easier to unit test the affected code.

Also, "populateVendorLegalName" and "formatVendorName" only appear to be used when processing a KFS/AVF-keyed vendor, not a PaymentWorks-keyed vendor. I will see if I can adapt them so that they can be used by this new code.